### PR TITLE
Do not use JIT metadata cache in AsyncGetCallTrace

### PR DESCRIPTION
**Pull request by gacholio**
ASGCT is called from a signal handler, so avoid memory allocation in the
stack walker.

[ci skip]

Signed-off-by: Graham Chapman <graham_chapman@ca.ibm.com>

### CODE REVIEW

**Comment by gacholio** [comment]
jenkins test sanity win,xlinux jdk8

File: runtime/j9vm/asgct.cpp [hunk 1]
```diff
@@ -120,6 +120,11 @@ protectedASGCT(J9PortLibrary *portLib, void *arg)
 		walkState.walkThread = currentThread;
 		walkState.skipCount = 0;
 		walkState.maxFrames = parms->depth;
+		/* JIT cache may allocate, which we don't want inside a signal handler,
+		 * Flag is cleared in the caller (so both the normal and exception cases
+		 * clear it).
+		 */
+		currentThread->jitArtifactSearchCache = (void*)((UDATA)currentThread->jitArtifactSearchCache | J9_STACKWALK_NO_JIT_CACHE);
```
Comment (tajila) [inline runtime/j9vm/asgct.cpp:127]: Can an allocation occur further up in `J9JITExceptionTable *metaData = jitConfig->jitGetExceptionTableFromPC(currentThread, rip);` as well?

Comment (gacholio) [reply runtime/j9vm/asgct.cpp:127]: No, the backend `jit_artifact_search` is strictly a read operation.

Comment (gacholio) [reply runtime/j9vm/asgct.cpp:127]: The stack walk is not allowed to fail due to memory allocation failure, so all of the allocations have a fallback - global buffer for stack/local map, and not using the metadata cache if it fails to allocate. We don't walk slots here, so the map case never occurs.

**Review by tajila — APPROVED** [review]

**Pull request merged**